marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/gradle/MarathonPlugin.kt

```diff
@@ -11,20 +11,31 @@ import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.logging.Logger
+import org.gradle.api.plugins.BasePlugin
 import org.gradle.api.plugins.JavaBasePlugin
 import org.gradle.kotlin.dsl.closureOf
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.getByName
 import java.io.File
+import java.nio.file.Paths
 
 class MarathonPlugin : Plugin<Project> {
 
     override fun apply(project: Project) {
         val logger = project.logger
         logger.info("Applying marathon plugin")
-        val extension: MarathonExtension = project.extensions.create("marathon", MarathonExtension::class.java, project)
+        project.extensions.create("marathon", MarathonExtension::class.java, project)
 
+        val rootProject = project.rootProject
+        synchronized(rootProject) {
```

Comment:
Do we need synchronized here?

Reply:
Do you know if gradle can use `apply` block on multiple projects at the same time? This sync is to have only one root project configuration basically